src/IJulia.jl

```diff
@@ -211,6 +211,36 @@ a subset of cell inputs/outputs to clear.
 clear_history
 
 #######################################################################
+# methods to print history or any subset thereof
+function history(io::IO, indices)
+	# since a range could be huge, intersect it with 1:n first
+    for n in intersect(indices, 1:IJulia.n)
+      if haskey(In, n)
+        print(In[n])
+      end
+    end
+end
+history(x) = history(Base.STDOUT, x)
+history() = history(1:n)
+
+# print the last x history entries when argument x is integer
+history(io::IO, x::Integer) =
+    invoke(history, Tuple{IO, Any}, io, n-x:n)
```

Comment:
Actually, I would define something more general, to match IPython:
```jl
function history(io::IO, indices::AbstractVector{Int})
    for n in sort!(unique(intersect(indices, 1:IJulia.n)))
        ...
    end
end

history(io::IO, x::Union{Integer,AbstractVector{Int}}...) = history(io, vcat(x...))
history(x...) = history(STDOUT, x...)
history(io::IO, x...) = throw(MethodError(history, (x...)))
```
This way you can do e.g. `history(3, 3:7, 9)` 

(I would have used `AbstractVector{T<:Integer}`, but some tricky dispatch issues arose, and it didn't seem worth resolving them.)